dali/pipeline/operators/resize/random_resized_crop.cu

```diff
@@ -44,7 +44,30 @@ void RandomResizedCrop<GPUBackend>::RunImpl(DeviceWorkspace * ws, const int idx)
   cudaStream_t old_stream = nppGetStream();
   nppSetStream(ws->stream());
 
-  for (int i = 0; i < batch_size_; ++i) {
+  /*
+   * workaround for out of bound acces from nppiResize
+   * add one more line in the last image
+   * it should not break anything as only one operator at the time can operate on the GPU
+   * and other should not keep any reference to the internals of the input tensor list
+  */
+  vector<Dims> new_shape(input.shape());
+  new_shape[batch_size_ - 1][1] += 1;
+  TensorList<GPUBackend> tmp_tensor_list;
+  auto &mutable_input = const_cast<TensorList<GPUBackend>&>(input);
+  tmp_tensor_list.ShareData(const_cast<TensorList<GPUBackend>*>(&input));
+  mutable_input.Resize(new_shape);
+  if (tmp_tensor_list.capacity() < input.capacity()) {
```

Comment:
I don't understand the logic here. You are resizing, then checking capacity and reverting the resize if the condition is not met. Can't you just check the condition before the first resize and skip it if necessary instead of resizing and rolling back?

Reply:
Done